### DISCUSSION TITLE
I would like to position the playhead in the center of a clip

### DISCUSSION THREAD

**Discussion by DaniDitu** [CommandPost/CommandPost#3209] [Lua Snippet Questions]
cp.apple.finalcutpro:launch() --open FCPX
cp.apple.finalcutpro:selectMenu({"Window", "Go To", "Timeline"}) -- select timeline
cp.apple.finalcutpro:doShortcut("SelectClip"):Now() -- select clip
local selectedClip = cp.apple.finalcutpro.timeline.selectedClips[1] -- error: [string "--cp.apple.finalcutpro:doShortcut("GoToTimeli..."]:7: attempt to call a nil value (method 'selectedClips')
local clipStart = selectedClip:getStartingTime()
local clipDuration = selectedClip:duration()
local clipCenter = clipStart + (clipDuration / 2)
cp.apple.finalcutpro:movePlayhead(clipCenter)

**Comment by latenitefilms** [comment]
Owww... good question. I don't think there's any easy way to do this, however, this might point you in the right direction:

https://github.com/CommandPost/CommandPost/blob/develop/src/plugins/finalcutpro/timeline/clipnavigation.lua#L38

I'll have a proper look at this when I get a chance.

**Comment by DaniDitu** [comment]
great, I managed to get the ClipNavigation plugin working, I wrote the command to run the command:
cp.apple.finalcutpro:launch() --richiamo il programma, serve altrimenti non funzione
cp.apple.finalcutpro:selectMenu({"Edit", "Select Clip"}) -- seleziona la clip in corrispondenza del playhead
cp.triggerAction("fcpx_cmds", {id="selectMiddleOfNextClipInSameLane"}) --Run Clip Navigation Select Next Clip
--cp.triggerAction("fcpx_cmds", {id="selectMiddleOfPreviousClipInSameLane"}) --Run Clip Navigation Select Previous Clip
